ponycheck/generator.pony

```diff
@@ -140,12 +140,92 @@ primitive Generators
         fun generate(rnd: Randomness): Set[T]^ =>
           let size = rnd.usize(0, max)
           let set = Set[T](size)
-          for i in Range(0, size) do
-            set.set(gen.generate(rnd))
-          end
+          set.union(
+            Iter[T^](gen.iter(rnd))
+              .take(size))
+          consume set
```

Comment:
i like to get more idiomatic with pony.
there have been > 10 times i wanted to have an exhaustive pony language reference with all those details.